PcapDotNet/src/PcapDotNet.Packets/Http/HttpRequestDatagram.cs

```diff
@@ -46,6 +46,18 @@ public override ILayer ExtractLayer()
                        Body = Body,
                    };
         }
+        
+        /// <summary>
+        /// An HTTP Request is valid if it contains a method, an URI, and a version.
+        /// </summary>
+        protected override bool CalculateIsValid()

```

Comment:
You're right. I didn't think of that because I reassemble the TCP datagrams before parsing the application layer.